modules/assim.sequential/R/downscale_function.R

```diff
@@ -1,67 +1,104 @@
 ##' @title Preprocess Data for Downscaling
 ##' @name SDA_downscale_preprocess
 ##' @author Sambhav Dixit
+##' @author David LeBauer
 ##'
-##' @param data_path Character. File path for .rds containing ensemble data.
-##' @param coords_path Character. File path for .csv file containing the site coordinates, with columns named "lon" and "lat".
-##' @param date Date. If SDA site run, format is yyyy/mm/dd; if NEON, yyyy-mm-dd. Restricted to years within the file supplied to 'data_path'.
-##' @param carbon_pool Character. Carbon pool of interest. Name must match the carbon pool name found within the file supplied to 'data_path'.
+##' @param ensemble_data Character or list. Either a file path to an `.rds` file containing ensemble data, or an in-memory list where each element is a data.frame indexed by date.
+##' @param site_coordinates Character, data.frame, or sf object. Either:
+##'   - A file path to a `.csv` containing site coordinates with columns `"id"`, `"lat"`, and `"lon"`.
+##'   - A `data.frame` with the same structure.
+##'   - An `sf` object with point geometries.
+##' @param date Date. If SDA site run, format is yyyy/mm/dd; if NEON, yyyy-mm-dd. Restricted to years within the file or object supplied to 'ensemble_data'.
```

Comment:
When this function was originally written I asked that this be fixed to just take in an actual Date. While you're fixing things could you fix that too.

Reply:
@mdietze I'm not sure if I fully understand, and how I can make sure not to brake existing code. 

What I have in mind is to 
- require the date parameter be a date object?
- if it is not a date object, try to convert from the two yyyymmdd options?

is that what you had in mind?

Reply:
@mdietze I'm not sure if I fully understand, and how I can make sure not to brake existing code. 

What I have in mind is to 
- require the date parameter be a Date object and update the docs accordingly and
- if it is not a date object, try to convert from the two yyyymmdd options? (already happens)

is that what you had in mind?